src/main/java/org/gridsuite/modification/server/entities/equipment/modification/LineSplitWithVoltageLevelEntity.java

```diff
@@ -54,32 +54,23 @@ public class LineSplitWithVoltageLevelEntity  extends ModificationEntity {
     @Column
     private String newLine2Name;
 
-    public LineSplitWithVoltageLevelEntity(String lineToSplitId, double percent,
-        VoltageLevelCreationEntity mayVoltageLevelCreation,
-        String existingVoltageLevelId, String bbsOrBusId,
-        String newLine1Id, String newLine1Name, String newLine2Id, String newLine2Name) {
+    public LineSplitWithVoltageLevelEntity(@NonNull LineSplitWithVoltageLevelInfos splitWithVoltageLevelInfos) {
         super(ModificationType.LINE_SPLIT_WITH_VOLTAGE_LEVEL);
-
-        this.lineToSplitId = lineToSplitId;
-        this.percent = percent;
-        this.mayVoltageLevelCreation = mayVoltageLevelCreation;
-        this.existingVoltageLevelId = existingVoltageLevelId;
-        this.bbsOrBusId = bbsOrBusId;
-        this.newLine1Id = newLine1Id;
-        this.newLine1Name = newLine1Name;
-        this.newLine2Id = newLine2Id;
-        this.newLine2Name = newLine2Name;
+        init(splitWithVoltageLevelInfos);
     }
 
     @Override
-    public void update(ModificationInfos modificationInfos) {
+    public void update(@NonNull ModificationInfos modificationInfos) {
         super.update(modificationInfos);
-        LineSplitWithVoltageLevelInfos splitWithVoltageLevelInfos = (LineSplitWithVoltageLevelInfos) modificationInfos;
+        init((LineSplitWithVoltageLevelInfos) modificationInfos);
+    }
+
+    private void init(LineSplitWithVoltageLevelInfos splitWithVoltageLevelInfos) {
```

Comment:
assignFields or assignValues or something else? I don't really like to having "init" inside "update"...